### PR TITLE
Add Array.from extension for scala 2.11 and 2.12

### PR DESCRIPTION
**Pull request by RustedBones**
Without the extension, code fails to compile for scala 2.11 and 2.12 with
```
 value from is not a member of object Array
     val a = Array.from(xs)
``` 

Add missing extension to the compat package

### CODE REVIEW

**Comment by SethTisue** [comment]
review by @scala/collections ?

File: compat/src/main/scala-2.11_2.12/scala/collection/compat/PackageShared.scala [hunk 1]
```diff
@@ -357,6 +360,11 @@ private[compat] trait PackageShared {
     new RandomExtensions(self)
 }
 
+class ArrayExtensions(private val fact: Array.type) extends AnyVal {
+  def from[A: ClassTag](source: TraversableOnce[A]): Array[A] =
+    fact.apply(source.toSeq: _*)
```
Comment (He-Pin) [inline compat/src/main/scala-2.11_2.12/scala/collection/compat/PackageShared.scala:365]: source match {
    case it: Iterable[A] => it.toArray[A]
    case _ => it.iterator.toArray[A]
  }

File: compat/src/main/scala-2.11_2.12/scala/collection/compat/PackageShared.scala [hunk 2]
```diff
@@ -357,6 +360,11 @@ private[compat] trait PackageShared {
     new RandomExtensions(self)
 }
 
+class ArrayExtensions(private val fact: Array.type) extends AnyVal {
```
Comment (He-Pin) [inline compat/src/main/scala-2.11_2.12/scala/collection/compat/PackageShared.scala:363]: final class?

**Review by SethTisue — APPROVED** [review]

**Comment by SethTisue** [comment]
Thank you!

**Comment by SethTisue** [comment]
this has now shipped: https://github.com/scala/scala-collection-compat/releases/tag/v2.13.0

**Pull request merged**